core/src/test/java/apoc/refactor/GraphRefactoringTest.java

```diff
@@ -994,5 +998,148 @@ public void testRefactorCategorizeNoDups() {
+        testCall(db, "MATCH (n:A), (m:B) WITH [n,m] as nodes CALL apoc.refactor.mergeNodes(nodes, {mergeRels: true, selfRel: true}) yield node return node",
+                (r) -> {
+                    Node node = (Node) r.get("node");
+                    List<Relationship> relationships = IteratorUtils.toList(node.getRelationships().iterator());
+
+                    assertEquals(3, relationships.size());
+                    relationships.sort(Comparator.comparingLong(Relationship::getStartNodeId)
+                            .thenComparingLong(Relationship::getEndNodeId));
+
+                    Relationship firstRel = relationships.get(0);
+                    assertEquals("A", firstRel.getStartNode().getLabels().iterator().next().name());
+                    assertEquals("A", firstRel.getEndNode().getLabels().iterator().next().name());
+                    assertEquals("KNOWS", firstRel.getType().name());
+
+                    Relationship secondRel = relationships.get(1);
+                    assertEquals("A", secondRel.getStartNode().getLabels().iterator().next().name());
+                    assertEquals("C", secondRel.getEndNode().getLabels().iterator().next().name());
+                    assertEquals("KNOWS", secondRel.getType().name());
+
+                    Relationship thirdRel = relationships.get(2);
+                    assertEquals("C", thirdRel.getStartNode().getLabels().iterator().next().name());
+                    assertEquals("A", thirdRel.getEndNode().getLabels().iterator().next().name());
+                    assertEquals("KNOWS", thirdRel.getType().name());
+                });
+    }
+
+    @Test
+    public void testMergeNodeShouldNotCancelOtherRelsWithSelfRelsFalseAndSingleNode() {
+        db.executeTransactionally("CREATE (a:A {a:'a'})-[:KNOWS]->(b:B {a:'b'})-[:KNOWS]->(c:C {a:'c'})\n" +
+                "WITH a,b,c CREATE (a)-[:KNOWS]->(a) WITH a,b,c CREATE (a)-[:KNOWS]->(c) WITH c,b CREATE (c)-[:KNOWS]->(b);");
+        testCall(db, "MATCH (n:A) WITH collect(n) as nodes CALL apoc.refactor.mergeNodes(nodes, {mergeRels: true, selfRel: false}) yield node return node",
+                (r) -> {
+                    Node node = (Node) r.get("node");
+                    List<Relationship> relationships = IteratorUtils.toList(node.getRelationships().iterator());
+
+                    assertEquals(3, relationships.size());
+                    relationships.sort(Comparator.comparingLong(Relationship::getStartNodeId)
+                            .thenComparingLong(Relationship::getEndNodeId));
+
+                    Relationship firstRel = relationships.get(0);
+                    assertEquals("A", firstRel.getStartNode().getLabels().iterator().next().name());
+                    assertEquals("A", firstRel.getEndNode().getLabels().iterator().next().name());
+                    assertEquals("KNOWS", firstRel.getType().name());
```

Comment:
why is this relationship preserved?

Reply:
Because in this case it merged only one node, this test is to demonstrate that the rels (besides nodes) will not change with 1 node.